src/component/style.ts

```diff
@@ -0,0 +1,87 @@
+import { Component } from "./component";
+
+export const globalStylesheets: { [key: string]: HTMLStyleElement } = {};
+
+export function registerSheet(id: string, css: string) {
+  const sheet = document.createElement("style");
+  sheet.innerHTML = processSheet(css);
+  globalStylesheets[id] = sheet;
+}
+
+/**
+ * Apply the stylesheets defined by the component. Note that we need to make
+ * sure all inherited stylesheets are applied as well, in a reverse order to
+ * ensure that <style/> will be applied to the DOM in the order they are
+ * included in the document. We then delete the `style` key from the constructor
+ * to make sure we do not apply it again.
+ */
+export function applyStyles(ComponentClass: typeof Component) {
+  const toApply: { [componentName: string]: string } = {};
+  while (ComponentClass && ComponentClass.style) {
+    if (ComponentClass.hasOwnProperty("style")) {
+      toApply[ComponentClass.name] = ComponentClass.style;
+      delete (ComponentClass as any).style;
+    }
+    ComponentClass = Object.getPrototypeOf(ComponentClass);
+  }
+  for (const [componentName, styleId] of Object.entries(toApply).reverse()) {
```

Comment:
reversing is new here. Any thoughts on that @ged-odoo ?

Reply:
toapply => list, push in it, then pop in while loop